src/chapters/orm-relationships/types-of-relationships.rst

```diff
@@ -0,0 +1,101 @@
+Types of Relationships
+======================
+
+Just as data in different database tables :ref:`can have relationships <one-to-many-sql>` to each other, so can classes and objects. In fact, relationships between objects will translate into relationships between rows in a database when using ORM. 
+
+This chapter introduces the tools needed to create meaningful relationships using ORM. Before considering how to do this with Spring Boot, however, let's consider the different types of relationships at a conceptual level. 
+
+For the examples below, we use four classes:
+
+- ``Event`` - A class representing a coding event.
+- ``EventCategory`` - A class representing categories of coding events.
+- ``EventDetails`` - A class that encapsulates details about a single event, such as description, contact email, location, and so on.
+- ``Tag`` - A piece of metadata labeling an event. You can think of these as topics that an event might include, such as Java, Spring, or JavaScript. An event can cover multiple topics, so it can have multiple tags.
+
+The first two of these are familiar to you from our ``coding-events`` app. ``EventDetails`` and ``Tag`` classes will be introduced later in this chapter.
+
+One-to-One
+----------
+
+.. index:: ! one-to-one
+
+The simplest type of relationship is the **one-to-one relationship**. This occurs when each instance of type A may be related to only one instance of type B, and vice versa.
+
+.. admonition:: Note
+
+   In the following examples, arrows point in the directly of the relationship. If A points to B, then you can say that A *knows about* B.
+
+.. figure:: figures/one-to-one.png
+   :alt: A single Event object on the left, with a double-sided arrow point to a single EventDetails object on the right
+   :width: 800px
+
+   A one-to-one relationship between an ``Event`` object and an ``EventDetails`` object
+
+An ``Event`` object should only have one collection of details, so it should only be related to one ``EventDetails`` object. Similarly, details about an event are specific to that event, so an ``EventDetails`` object should only be related to one ``Event`` object.
+
+.. admonition:: Examples
+
+   The following pairs of things generally have one-to-one relationships:
+
+   #. People / driver's licenses
+   #. States / capital cities
+   #. iPhones / serial numbers
+
+   It is not *required* that each instance of type A be related to an instance of type B. For example, a person may not have a driver's license.
+
+One-to-Many and Many-to-One
+---------------------------
+
+.. index:: ! one-to-many
+
+A **one-to-many** relationship occurs when each instance of type A may be related to more than one instance of type B, but each instance of B can only be related to a single instance of type A.
+
+.. figure:: figures/one-to-many.png
```

Comment:
I'm unclear about the use of one-directional vs. bi-directional arrows in this figure (and the many-to-many figure).  With the SQL foreign keys, does each item in the _many_ know about the _one_?

For the ``Tag`` example in the many-to-many figure, if one tag can be related to many different events, then does the tag "know about" each event?  If so, then this would imply bi-directional arrows.  If not, then why does an ``Event`` (also a _many_) know about each related tag but not the other way around?

Reply:
In Java, relationships are only bi-direction if specific to be so. So in these examples, I have leaned on emphasizing the direction of the relationship to make it a bit more clear.

The way in which one object "knows about" another one is through a reference (aka a field). So, an `Event` may have a `Tag` field, but the `Tag` object doesn't necessarily know which events "know about" it.

Reply:
OK.  Could you add that clarification into the text?